contracts/interchain-token-service/src/contract.rs

```diff
@@ -315,17 +318,7 @@ impl InterchainTokenService {
     ) -> Result<Bytes, ContractError> {
         // Note: ITS Hub chain as the actual destination chain for the messsage isn't supported
         ensure!(
-            destination_chain != Self::its_hub_chain_name(env),
-            ContractError::UntrustedChain
-        );
-
-        let Some(destination_address) = Self::trusted_address(env, destination_chain.clone())
-        else {
-            return Err(ContractError::UntrustedChain);
-        };
-
-        ensure!(
-            destination_address == Self::its_hub_routing_identifier(env),
+            Self::trusted_chain(env, destination_chain.clone()),
```

Comment:
this change means the configuration is different, i.e we should avoid trusting the its hub chain name now, since it's not a true source/destination. The ITS hub address is read separately. Document this appropriately in the operational script so it's clear to chain ops